webapp/src/app.tsx

```diff
@@ -67,7 +67,16 @@ interface IAppState {
     showParts?: boolean;
 }
 
+interface ExternalMessageData {
+    messageType: string;
+    args: any
+}
+interface ExternalMessageResult {
+    error?: any;
+    result?: any;
+}
 
+let isElectron = /[?&]electron=1/.test(window.location.href);
```

Comment:
Please consider a more generic type until a specific need for the label isElectron.
For example notBrowser to indicate another type of usage.
Perhaps coupled with a --not-browser option instead of a --electron